### PR TITLE
Fix lifecycle introduced by #936

### PR DESCRIPTION
**Pull request by tomwilkie**
#936 was pretty wrong; this change reduces run time for the test correctly.

Signed-off-by: Tom Wilkie <tom.wilkie@gmail.com>

### CODE REVIEW

File: pkg/ingester/lifecycle_test.go [hunk 1]
```diff
@@ -123,8 +120,11 @@ func TestIngesterTransfer(t *testing.T) {
 		}, nil
 	}
 
-	// Now stop the first ingester
+	// Now stop the first ingester, and wait for the second ingester to become ACTIVE.
 	ing1.Shutdown()
+	poll(t, 10*time.Second, ring.ACTIVE, func() interface{} {
```
Comment (csmarchbanks) [inline pkg/ingester/lifecycle_test.go]: It seems like this poll is very necessary. Was this test just passing by luck beforehand?

Comment (csmarchbanks) [reply pkg/ingester/lifecycle_test.go]: Ohh wait, looks like Shutdown will block until all the chunks are transferred.

Comment (tomwilkie) [reply pkg/ingester/lifecycle_test.go]: Yes not very necessary, just there when I was debugging.  Can remove if you like?

Comment (csmarchbanks) [reply pkg/ingester/lifecycle_test.go]: It seems like a reasonable check to me, I would say leave it in.

**Pull request merged**